src/components/Gradebook/index.jsx

```diff
@@ -22,48 +22,56 @@ export default class Gradebook extends React.Component {
     this.state = {
       filterValue: '',
       modalOpen: false,
-      modalModel: [{}],
-      updateVal: 0,
+      adjustedGradeValue: 0,
       updateModuleId: null,
       updateUserId: null,
+      reasonForChange: '',
     };
     this.fileFormRef = React.createRef();
     this.fileInputRef = React.createRef();
+    this.myRef = React.createRef();
   }
 
   componentDidMount() {
     const urlQuery = queryString.parse(this.props.location.search);
     this.props.getRoles(this.props.courseId, urlQuery);
+    this.overrideReasonInput.focus();
   }
 
-  setNewModalState = (userEntry, subsection) => {
-    let adjustedGradePossible = '';
-    let currentGradePossible = '';
-    if (subsection.attempted) {
-      adjustedGradePossible = ` / ${subsection.score_possible}`;
-      currentGradePossible = `/${subsection.score_possible}`;
-    }
-    this.setState({
-      modalModel: [{
-        username: userEntry.username,
-        currentGrade: `${subsection.score_earned}${currentGradePossible}`,
-        adjustedGrade: (
-          <span>
-            <input
-              style={{ width: '25px' }}
-              type="text"
-              onChange={event => this.setState({ updateVal: event.target.value })}
-            />{adjustedGradePossible}
-          </span>
-        ),
-        assignmentName: `${subsection.subsection_name}`,
-      }],
-      modalOpen: true,
-      updateModuleId: subsection.module_id,
-      updateUserId: userEntry.user_id,
-    });
+  onChange(e) {
+    this.setState({ [e.target.name]: e.target.value });
   }
 
+   setNewModalState = (userEntry, subsection) => {
+     this.props.fetchGradeOverrideHistory(
+       subsection.module_id,
+       userEntry.user_id,
+     );
+
+     let adjustedGradePossible = '100';
+     if (subsection.attempted) {
+       adjustedGradePossible = ` / ${subsection.score_possible}`;
+     }
+     const options = { year: 'numeric', month: 'long', day: 'numeric' };
+     const timeOptions = { hour: '2-digit', minute: '2-digit' };
+     const dt = new Date();
+     const todaysdate = `${dt.toLocaleDateString('en-US', options)} at ${dt.toLocaleTimeString('en-US', timeOptions)}`;
```

Comment:
Use the formatDate Function that has been moved to a utility file to do this formatting. That comment was #3000